cmd/docker/buildx.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"os"
+	"strconv"
+
+	pluginmanager "github.com/docker/cli/cli-plugins/manager"
+	"github.com/docker/cli/cli/command"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+const (
+	buildxPluginName = "buildx"
+	buildxErrorMsg   = `ERROR: Missing builder component buildx. To install buildx, see
+       https://docs.docker.com/buildx/working-with-buildx/#install. You
+       can also fallback to the legacy builder by setting DOCKER_BUILDKIT=0`
+)
+
+type buildxError struct {
+	err error
+}
+
+func (e *buildxError) Error() string {
+	if pluginmanager.IsNotFound(e.err) {
+		return errors.New(buildxErrorMsg).Error()
+	}
+	return errors.Errorf("%v\n\n%s", e.err, buildxErrorMsg).Error()
+}
+
+func processBuildx(dockerCli command.Cli, cmd *cobra.Command, args, osArgs []string) ([]string, []string, error) {
+	// check DOCKER_BUILDKIT env var is present and
+	// if not assume we want to use buildx
+	if v, ok := os.LookupEnv("DOCKER_BUILDKIT"); ok {
+		enabled, err := strconv.ParseBool(v)
+		if err != nil {
+			return args, osArgs, errors.Wrap(err, "DOCKER_BUILDKIT environment variable expects boolean value")
+		}
+		if !enabled {
+			return args, osArgs, nil
+		}
+	}
+
+	// if a builder alias is defined but not buildx, skip forwarding
+	aliasMap := dockerCli.ConfigFile().Aliases
+	if v, ok := aliasMap[keyBuilderAlias]; ok && v != buildxPluginName {
+		return args, osArgs, nil
+	}
```

Comment:
@tonistiigi Now buildx forwarding by default is skipped if builder alias is defined but not buildx as discussed.